sql/sql_zip_dict.cc

```diff
@@ -94,8 +98,19 @@ int mysql_create_zip_dict(THD* thd, const char* name, ulong name_len,
         error= ER_READ_ONLY_MODE;
         my_error(error, MYF(0));
         break;
+      case HA_CREATE_ZIP_DICT_OUT_OF_MEMORY:
```

Comment:
The testcase handles HA_CREATE_ZIP_DICT_TOO_MANY_CONCURRENT_TRXS, how did you come up with the other two - by code review? Is it worth to add debug injection tests for them too?